.github/workflows/smoketest.yml

```diff
@@ -0,0 +1,391 @@
+    # 4) get submodules
+    # 5) copy src-cache files into place
+    # 6) touch all files so that 'old' [before original make timestamp], i.e. so behind objects in cache
+    # 7) checkout pr branch - ensuring only pr files that differ are updated (and hence will rebuild)
+    # 8) list updated files (for tracing only)
+    - name: Cache hit prep
+      if: steps.cache.outputs.cache-hit != 'true' && steps.check-cache.outputs.cache-exists == 'true'
+      run: |
+        echo Attempting to use existing cache
+        echo Fetching branch: ${{ steps.vars.outputs.branch }}
+        git clone --branch ${{ steps.vars.outputs.branch }} -n https://github.com/${{ steps.vars.outputs.repository }} src
+        mv build-cache build
+        cd src
+        if [[ "commit" == "$(git cat-file -t $(cat ../build/cache-sha))" ]]; then \
+          echo Checking out cache SHA: $(cat ../build/cache-sha); \
+          git checkout $(cat ../build/cache-sha); \
+        else \
+          echo git diff failed \(SHA $(cat ../build/cache-sha) missing?\). Using base_ref; \
+          git checkout ${{ steps.vars.outputs.base_ref }}; \
+        fi
+        git submodule update --init --recursive --jobs 4
+        cd ../src-cache
+        echo Cached source tree files:
+        find . -type f
+        echo ================
+        find . -type f | cpio -p -dum ../src
+        cd ../src
+        find . -type f -exec touch -r ../build/cache-timestamp {} +
+        git checkout ${{ steps.vars.outputs.branch }}
+        git submodule update --init --recursive --jobs 4
+        echo Changed files \(from SHA: $(cat ../build/cache-sha)\):
+        find -name '.git*' -prune -o -newer ../build/cache-timestamp -type f -print
+
+    - name: Cache miss checkout
+      if: steps.cache.outputs.cache-hit != 'true' && steps.check-cache.outputs.cache-exists != 'true'
+      uses: actions/checkout@v2
+      with:
+        submodules: recursive
+        path: src
+
+    - name: Cache miss prep
+      if: steps.cache.outputs.cache-hit != 'true' && steps.check-cache.outputs.cache-exists != 'true'
+      run: |
+        rm -rf build src-cache
+        mkdir build
+        mkdir src-cache
+
+    # Subsequent cache hit builds use cache-timestamp to ensure all sources are marked 'old',
+    # except those changed between the cache SHA and the SHA being built.
+    - name: Build
+      if: steps.cache.outputs.cache-hit != 'true'
+      run: |
+        cd build
+        if [[ ! -f cache-timestamp ]]; then touch cache-timestamp; fi
+        touch ../cmake-timestamp
+        ../cmake/cmake ../src -Wno-dev -DRUNTIME_USER=$(whoami) -DRUNTIME_GROUP=$(id -gn) -DDESTDIR=$(realpath ..)/install -DINCLUDE_PLUGINS=1 -DCMAKE_BUILD_TYPE=RelWithDebInfo -DUSE_PYTHON2=0 -DUSE_PYTHON3=1 -DSUPPRESS_SPARK=1 -DUSE_CPPUNIT=1 -DUSE_CASSANDRA=Off -DCMAKE_CXX_FLAGS="-DBUILD_TAG=\\\"dummytag\\\" -DBUILD_VERSION_MAJOR=1 -DBUILD_VERSION_MINOR=1 -DBUILD_VERSION_POINT=0" -DBUILD_TAG="smoketest"
```

Comment:
The reason for always using the same tag is to maximise effectiveness of the cache? Would a comment help make that clear?

Reply:
yes, otherwise a PR tag would be based on the SHA, and therefore change constantly, and cause quite a lot of rebuilding.
I'll add a comment.